pkg/kubelet/status/manager.go

```diff
@@ -323,8 +323,14 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 	// TODO: make me easier to express from client code
 	pod, err := m.kubeClient.Pods(status.podNamespace).Get(status.podName)
 	if errors.IsNotFound(err) {
-		glog.V(3).Infof("Pod %q (%s) was deleted on the server", status.podName, uid)
-		m.deletePodStatus(uid)
+		glog.V(3).Infof("Pod %q (%s) does not exist on the server", status.podName, uid)
+		// Note that we do not delete the pod status because the pod to update may
+		// be a static pod whose mirror pod is in the process of recreation.
+		// Deleting the status without clearing the corresponding
+		// apiStatusVersions entry may lead to wrong versioning and stale status.
```

Comment:
> I think there are too many other ways that the Pod can change to make that useful in this context. I was talking about adding a resourceVersion to the PodStatus object. I imagine that breaks some API conventions, but it would simplify a lot in this case.

In that case, PodStatus should be a separate resource, and kubelet should watch the PodStatus in addition to the Pod objects. This would increase the watch connection and probably affect the cluster scalability. I am doubtful that adding a version in PodStatus will be well-received.


Changed the TODO comment based on the suggestion.